"<:div><script>alert()<:script>{{1+1}}.java

```diff
@@ -0,0 +1,68 @@
+package cucumber.helpers.shared;
+
+import com.amazonaws.services.dynamodbv2.LockItem;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class PackageLocksHelper {
+    private static final Logger log = LogManager.getLogger(PackageLocksHelper.class);
+    private final PackageLock packageLock;
+    private final String stage;
+    private final Map<String, LockItem> lockedPackages = new ConcurrentHashMap<>();
+
+    public PackageLocksHelper(PackageLock packageLock, String stage) {
+        this.packageLock = packageLock;
+        this.stage = stage.substring(0, 1).toUpperCase() + stage.substring(1).toLowerCase();
+    }
+
+    public String lockGitFarmPackageWithStage(String packagePrefix) {
+        int count = this.packageLock.getCountForPackage(packagePrefix);
+        List<String> packagePool = IntStream.rangeClosed(1, count)
+                .mapToObj(String::valueOf)
+                .map(i -> packagePrefix + "-" + stage + i)
+                .collect(Collectors.toList());
+        return this.lockGitFarmPackage(packagePool);
+    }
+
+    public String lockGitFarmPackage(String packagePrefix) {
+        int count = this.packageLock.getCountForPackage(packagePrefix);
+        List<String> packagePool = IntStream.rangeClosed(1, count)
+                .mapToObj(String::valueOf)
+                .map(i -> packagePrefix + i)
+                .collect(Collectors.toList());
+        return this.lockGitFarmPackage(packagePool);
+    }
+
+    public String lockGitFarmPackage(List<String> packagePool) {
+        try {
+            LockItem lockPackage = this.packageLock.getLockedPackage(packagePool);
+            String packageName = lockPackage.getSortKey().get();
+
+            this.lockedPackages.put(packageName, lockPackage);
+            log.info("A lock is acquired for GitFarm package " + packageName);
+            return packageName;
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

## Problem\n\n You are using a `ConcurrentHashMap`, but your usage of `containsKey()` and `get()` may not be thread-safe at lines: [56 and 59](https://code.amazon.com/packages/testpentest/blobs/b716e04cb52b782205e17839241c34a6db0fb9b7/--/\&#34;\u003c:div\u003e\u003cscript\u003ealert()\u003c:script\u003e{{1+1}}.java#L56-L59).\n\n In between the check and the `get()` another thread can remove the key and the `get()` will return `null`.\n\n The removes that can remove the key are at lines: [60](https://code.amazon.com/packages/testpentest/blobs/b716e04cb52b782205e17839241c34a6db0fb9b7/--/\&#34;\u003c:div\u003e\u003cscript\u003ealert()\u003c:script\u003e{{1+1}}.java#L60-L60) and [66](https://code.amazon.com/packages/testpentest/blobs/b716e04cb52b782205e17839241c34a6db0fb9b7/--/\&#34;\u003c:div\u003e\u003cscript\u003ealert()\u003c:script\u003e{{1+1}}.java#L66-L66).\n\n## Fix\n\n Consider calling `get()`, checking instead of your current check if the returned object is `null`, and then using that object only, without calling `get()` again.\n\n## More info\n\n Example fix: [View an example on GitHub](https://github.com/apache/hadoop/blob/f16cf877e565084c66bc63605659b157c4394dc8/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/S3Guard.java#L302-L304) (external link).\n\n